core/src/main/scala/Config.scala

```diff
@@ -530,18 +535,73 @@ object Config {
         )((a,b,c,d,e,g) => {
           val splunk = readSplunk
           val loggingSidecar = readLoggingImage
-          val uri = org.http4s.Uri.fromString(a).toOption.yolo(s"nomad.endpoint -- $a -- is an invalid Uri")
+          val uri = Uri.fromString(a).toOption.yolo(s"nomad.endpoint -- $a -- is an invalid Uri")
           Infrastructure.Nomad(uri,b,c,d,e,loggingSidecar,g,splunk)
         })
     }
 
-    /*
-     * Datacenters currently only support one scheduler
-     */
-    def readScheduler(kfg: KConfig, proxy: Option[Infrastructure.ProxyCredentials]): Option[SchedulerOp ~> Task] =
-      readNomadInfrastructure(kfg.subconfig("nomad"))
+    def readKubernetesInfrastructure(kfg: KConfig): Option[Infrastructure.Kubernetes] = {
+      (kfg.lookup[String]("endpoint") |@| kfg.lookup[Duration]("timeout")) { (endpoint, timeout) =>
+        val uri = Uri.fromString(endpoint).toOption.yolo(s"kubernetes.endpoint -- $endpoint -- is an invalid Uri")
+        Infrastructure.Kubernetes(uri, timeout)
+      }
+    }
+
+    def readNomadScheduler(kfg: KConfig): Option[SchedulerOp ~> Task] =
+      readNomadInfrastructure(kfg)
         .map(n => new scheduler.NomadHttp(nomadcfg, n, http4sClient(n.timeout)))
 
+    // Create a X509TrustManager that trusts a whitelist of certificates, similar to `curl --cacert`
+    def cacert(certs: Array[X509Certificate]): X509TrustManager =
+      new X509TrustManager {
+        def getAcceptedIssuers(): Array[X509Certificate] = certs
+        def checkClientTrusted(certs: Array[X509Certificate], authType: String): Unit = ()
+        def checkServerTrusted(certs: Array[X509Certificate], authType: String): Unit = ()
+      }
+
+    def getKubernetesPodCert(): Option[SSLContext] = {
+      // Auto-mounted at this path for pods
+      // https://kubernetes.io/docs/tasks/access-application-cluster/access-cluster/#accessing-the-api-from-a-pod
+      val certBundle = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+
+      val is = new FileInputStream(certBundle)
+      val x509Cert: Option[X509Certificate] = try {
+        val cf = CertificateFactory.getInstance("X.509")
+        cf.generateCertificate(is) match {
+          case c: X509Certificate => Some(c)
+          case _                  => None
+        }
+      } catch {
+        case NonFatal(_) => None
+      } finally {
+        is.close()
+      }
+
+      x509Cert.map { cert =>
+        val trustManager = cacert(Array(cert))
+        val sslContext = SSLContext.getInstance("TLS")
+        sslContext.init(null, Array(trustManager), new SecureRandom())
```

Comment:
Doesn't this make a new secure random for every request? It's not clear if this code block is called frequently or not.

Reply:
Appears so yeah, but it should only be called once on startup I think since (currently) it's only being used to create the K8s scheduler interpreter. I could move it out but I'm not sure what the security or mutability ramifications of that are.